### PR TITLE
Fix broken X-Scale (2)

### LINKED ISSUE #382: Step size incorrect when reading from yaml
The X-scale "Step Size" value does not appear to be set correctly when reading in a yaml input file.  Previously saving the file with log binning checked and a step size of 0.03 entered, re-loading the file reverts to a value of -0.01 and log binning _unchecked_.

### PR DESCRIPTION
**Pull request by TerriLouise**
Closes #382.

### CODE REVIEW

**Review by TerriLouise — COMMENTED** [review]
Second commit to fix previously introduced linting error

**Comment by TerriLouise** [comment]
When loading YAML input files, X-scale parameters read in correctly.

**Comment by trisyoungs** [comment]
From Jared's original PR (#396) for posterity:

I believe the issue outlined in https://github.com/disorderedmaterials/GudPy/issues/382 was related to two different issues:
- Changing the XMin never worked, due to a typo.
- Switching between scales was being handled incorrectly leading to incorrectly saving values.
This wasn't noticed, in this case, when saving to a text file because that stores the XStep as -0.01 when logarithmic binning is checked.

File: gudpy/gui/widgets/slots/instrument_slots.py [hunk 1]
```diff
@@ -989,17 +989,20 @@ def handleDSpacingToggled(self, state):
...
-            if not self.widgetsRefreshing:
+        if not self.widgetsRefreshing:
+            if state:
+                values = [
+                    widget.value()
+                    for widget in self.scales[Scales.D_SPACING][1:]
+                ]
+
+                (
+                    self.instrument.Xmin,
```
Comment (trisyoungs) [inline gudpy/gui/widgets/slots/instrument_slots.py:1000]:
```suggestion
                    self.instrument.XMin,
```

File: gudpy/gui/widgets/slots/instrument_slots.py [hunk 2]
```diff
@@ -1016,17 +1019,20 @@ def handleWavelengthToggled(self, state):
...
-            if not self.widgetsRefreshing:
+        if not self.widgetsRefreshing:
+            if state:
+                values = [
+                    widget.value()
+                    for widget in self.scales[Scales.WAVELENGTH][1:]
+                ]
+
+                (
+                    self.instrument.Xmin,
```
Comment (trisyoungs) [inline gudpy/gui/widgets/slots/instrument_slots.py:1030]:
```suggestion
                    self.instrument.XMin,
```

File: gudpy/gui/widgets/slots/instrument_slots.py [hunk 3]
```diff
@@ -1040,17 +1046,20 @@ def handleWavelengthStepChanged(self, value):
...
-            if not self.widgetsRefreshing:
+        if not self.widgetsRefreshing:
+            if state:
+                values = [
+                    widget.value()
+                    for widget in self.scales[Scales.ENERGY][1:]
+                ]
+
+                (
+                    self.instrument.Xmin,
```
Comment (trisyoungs) [inline gudpy/gui/widgets/slots/instrument_slots.py:1057]:
```suggestion
                    self.instrument.XMin,
```

File: gudpy/gui/widgets/slots/instrument_slots.py [hunk 4]
```diff
@@ -1064,15 +1073,20 @@ def handleEnergyStepChanged(self, value):
...
-            if not self.widgetsRefreshing:
+        if not self.widgetsRefreshing:
+            if state:
+                values = [
+                    widget.value()
+                    for widget in self.scales[Scales.TOF][1:]
+                ]
+
+                (
+                    self.instrument.Xmin,
```
Comment (trisyoungs) [inline gudpy/gui/widgets/slots/instrument_slots.py:1084]:
```suggestion
                    self.instrument.XMin,
```

**Review by trisyoungs — CHANGES_REQUESTED** [review]
Some small changes to absorb before merging.

**Pull request merged**